core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -390,3 +390,214 @@ message SQLExecutionUIData {
   repeated int64 stages = 11;
   map<int64, string> metric_values = 12;
 }
+
+message StageDataWrapper {
+  StageData info = 1;
+  repeated int64 job_ids = 2;
+  map<string, int64> locality = 3;
+}
+
+message TaskData {
+  int64 task_id = 1;
+  int32 index = 2;
+  int32 attempt = 3;
+  int32 partition_id = 4;
+  int64 launch_time = 5;
+  optional int64 result_fetch_start = 6;
+  optional int64 duration = 7;
+  string executor_id = 8;
+  string host = 9;
+  string status = 10;
+  string task_locality = 11;
+  bool speculative = 12;
+  repeated AccumulableInfo accumulator_updates = 13;
+  optional string error_message = 14;
+  optional TaskMetrics task_metrics = 15;
+  map<string, string> executor_logs = 16;
+  int64 scheduler_delay = 17;
+  int64 getting_result_time = 18;
+}
+
+message StageData {
+  enum StageStatus {
```

Comment:
If `StageStatus` is defined outside, the error message is as follows：
<img width="508" alt="image" src="https://user-images.githubusercontent.com/15246973/209742331-cf5e5af7-3321-41e2-b094-28193e397274.png">

Then If `StageStatus` is defined as follows:
enum StageStatus {
  STAGE_STATUS_UNSPECIFIED = 0;
  STAGE_STATUS_ACTIVE = 1;
  STAGE_STATUS_COMPLETE = 2;
  STAGE_STATUS_FAILED = 3;
  STAGE_STATUS_PENDING = 4;
  STAGE_STATUS_SKIPPED = 5;
}

The Code of `Serializer` and `Deerializer` will be very ugly!
Will have to handle the operations of adding prefix and deleting prefix.

Reply:
In fact, the enum definition of JobExecutionStatus seems more reasonable in JobData ?

Reply:
fine to me

Reply:
JobExecutionStatus is used in `SQLExecutionUIData`. So it can't be moved into `JobData`

Reply:
As described in  https://github.com/apache/spark/pull/39270/files, `UNSPECIFIED` in `StageStatus`  should change to `STAGE_STATUS_UNSPECIFIED`

Reply:
A.If the enum definition of `JobExecutionStatus` in JobData
<img width="224" alt="image" src="https://user-images.githubusercontent.com/15246973/209922039-617b3993-b6c2-48dc-a24f-fdb343585c68.png">

SQLExecutionUIData can use it as follows:
<img width="414" alt="image" src="https://user-images.githubusercontent.com/15246973/209922101-a2504145-87a3-46f7-b918-6d4363773d17.png">

B.Other choices for the definition of `JobExecutionStatus` as follows:
<img width="233" alt="image" src="https://user-images.githubusercontent.com/15246973/209922683-296a5c81-daa8-48ad-8bd2-8ea4682953e5.png">
<img width="699" alt="image" src="https://user-images.githubusercontent.com/15246973/209922716-51945000-6654-4283-affb-37e2dbd5978e.png">
<img width="390" alt="image" src="https://user-images.githubusercontent.com/15246973/209922853-d0d1affa-6c94-454a-883a-b87551c7ddb0.png">

Otherwise, I'm afraid there are other enumeration names that will conflict. @gengliangwang

Reply:
Ok, Let us follow code style(https://developers.google.com/protocol-buffers/docs/style#enums):
<img width="860" alt="image" src="https://user-images.githubusercontent.com/15246973/209947610-c586a3af-479b-4453-820b-7b853dff5009.png">

Reply:
New pr for JobExecutionStatus: https://github.com/apache/spark/pull/39286
@gengliangwang @LuciferYang